File: src/typescript/api/default-api/set-endpoint.ts
import { createHiveChain } from '@hiveio/wax';

const chain = await createHiveChain();

// Custom endpoint URL for database_api
chain.api.database_api.endpointUrl = "https://best.honey.provider";

chain.endpointUrl = "https://api.hive.blog"; // This is default for all APIs

console.log(chain.api.database_api.endpointUrl);

File: src/rust/api/default_api/set_endpoint.rs
use wax::prelude::*;

#[tokio::main]
async fn main() -> Result<(), Box<dyn std::error::Error>> {
    let chain = create_hive_chain(None)?;

    // Custom endpoint URL for database_api
    chain
        .api()
        .database_api
        .set_endpoint_url(Some("https://best.honey.provider".into()));

    chain.set_endpoint_url("https://api.hive.blog")?; // This is default for all APIs

    println!("{}", chain.endpoint_url());

    Ok(())
}
